sale_product_set_sale_by_packaging/models/product_set_line.py

```diff
@@ -24,3 +30,53 @@ def _check_sell_only_by_packaging_err_msg(self, lines):
         return _(
             "The following product(s) can be sold only by packaging: \n   %s"
         ) % ", ".join(lines.mapped("product_id.display_name"))
+
+    @api.model
+    def cron_check_packaging(self):
+        """Ensure lines are tied to the right packaging.
+
+        When
+
+        * packaging cannot be sold anymore
+        * lines are tied to a product that can be sold only by packaging
+          and have no packaging
+
+        change the packaging on the line to the 1st saleable packaging.
```

Comment:
If you want to make it optional you can turn off the cron. To me it's more important to keep data consistent rather than rely on user action. Moreover, this is not changing a line on a sale order. The user will be able to change it on product set or on sale order before confirming anyway.